bpfprogs/nfconfig.go

```diff
@@ -759,19 +760,32 @@ func (c *NFConfigs) Deploy(ifaceName, HostName string, bpfProgs *models.BPFProgr
 
 // DeployeBPFPrograms - Starts eBPF programs on the node if they are not running
 func (c *NFConfigs) DeployeBPFPrograms(bpfProgs []models.L3afBPFPrograms) error {
+	var combinedError error
+
 	for _, bpfProg := range bpfProgs {
 		if err := c.Deploy(bpfProg.Iface, bpfProg.HostName, bpfProg.BpfPrograms); err != nil {
 			if err := c.SaveConfigsToConfigStore(); err != nil {
-				return fmt.Errorf("deploy eBPF Programs failed to save configs %w", err)
+				if combinedError == nil { // saving all the errors for c.Deploy c.SaveConfigsToConfigStore to combinedError instead of returning right away.
+					combinedError = fmt.Errorf("deploy eBPF Programs failed to save configs %w", err)
+				} else {
+					combinedError = fmt.Errorf("%v; %v", combinedError, fmt.Errorf("deploy eBPF Programs failed to save configs %w", err))
+				}
+			}
+			if combinedError == nil {
+				combinedError = fmt.Errorf("deploy eBPF Programs failed to save configs %w", err)
+			} else {
+				combinedError = fmt.Errorf("%v; %v", combinedError, fmt.Errorf("failed to deploy BPF program on iface %s with error: %w", bpfProg.Iface, err))
```

Comment:
Can we remove this SaveConfigsToConfigStore(); call ? we have to save config at the end with the good state only.

Reply:
Removed the SaveConfigsToConfigStore() call